tests/integration_tests/build/test_dependencies.py

```diff
@@ -24,4 +24,10 @@ def test_licenses():
         os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../Cargo.toml")
     )
 
-    cargo("deny", f"--manifest-path {toml_file} check licenses bans")
+    res = cargo("deny", f"--manifest-path {toml_file} check licenses bans")
+    # if there is an incompatible license which is not copyleft and not added
+    # explicitly in the deny list of deny.toml then, it will be shown as part of
+    # warnings. Make sure we review all warnings related to licenses and take
+    # appropriate actions (i.e. either add them to allow list of create an exception here)
+    license_res = [line for line in res.stderr.split("\n") if "licenses" in line]
+    assert not license_res
```

Comment:
```suggestion
    res = cargo("deny", f"--manifest-path {toml_file} check licenses bans")
    # if there is an incompatible license which is not copyleft and not added
    # explicitly in the deny list of deny.toml then, it will be shown as part of
    # warnings. Make sure we review all warnings related to licenses and take
    # appropriate actions (i.e. either add them to allow list of create an exception here)
    _, stdout, stderr = [line for line in res.stderr.split("\n") if "licenses" in line]
    assert not stderr
    assert stdout == "licenses ok"  # not sure if the coloring might mess things up here
```